compiler/Makefile

```diff
@@ -10,54 +10,40 @@ libdir   ?= $(exec_prefix)/lib
 .PHONY: default
 default: stage$(STAGE)
 
-.PHONY: runtime/libskip_runtime64.a
-runtime/libskip_runtime64.a:
-	$(MAKE) -C runtime $(@F)
-
-.PHONY: runtime/libbacktrace/.libs/libbacktrace.a
-runtime/libbacktrace/.libs/libbacktrace.a:
-	$(MAKE) -C runtime libbacktrace
-
-.PHONY: runtime/libskip_runtime32.bc
-runtime/libskip_runtime32.bc:
-	$(MAKE) -C runtime $(@F)
-
-bootstrap/%.ll: bootstrap/%.ll.gz
-	gunzip -c $< > $@
-
 stage%:
 	mkdir -p $@
-	$(MAKE) $@/bin/skc $@/bin/skfmt $@/bin/skargo $@/lib/libskip_runtime64.a $@/lib/libbacktrace.a $@/lib/libskip_runtime32.bc $@/lib/skip_preamble64.ll $@/lib/skip_preamble32.ll
+	$(MAKE) $@/bin/skc $@/bin/skfmt $@/bin/skargo $@/lib/libstd.sklib $@/lib/libstd-wasm32-unknown-unknown.sklib
 
-stage%/lib/libskip_runtime64.a: runtime/libskip_runtime64.a
+stage%/lib/libstd.sklib: bootstrap/libstd.sklib
 	mkdir -p $(@D)
 	cp $< $@
 
-stage%/lib/libbacktrace.a: runtime/libbacktrace/.libs/libbacktrace.a
+# TODO: Get rid of this once skargo handles building sklibs.
+stage%/lib/libstd-wasm32-unknown-unknown.sklib: bootstrap/libstd-wasm32-unknown-unknown.sklib
 	mkdir -p $(@D)
 	cp $< $@
 
-stage%/lib/libskip_runtime32.bc: runtime/libskip_runtime32.bc
-	mkdir -p $(@D)
-	cp $< $@
+bootstrap/%.ll: bootstrap/%.ll.gz
+	gunzip -c $< > $@
 
-stage%/lib/skip_preamble64.ll: preamble/preamble64.ll
-	mkdir -p $(@D)
-	cp $< $@
+.PHONY: bootstrap/libskip_runtime64.a bootstrap/libbacktrace.a
+bootstrap/libskip_runtime64.a:
+bootstrap/libbacktrace.a:
+	OUT_DIR=$(realpath ./bootstrap) $(MAKE) -C ../prelude
 
-stage%/lib/skip_preamble32.ll: preamble/preamble32.ll
-	mkdir -p $(@D)
-	cp $< $@
+.PHONY: bootstrap/libstd.sklib
+bootstrap/libstd.sklib:
+	PATH=$(realpath stage0/bin/):$(PATH) OUT_DIR=$(realpath ./bootstrap) $(MAKE) -C ../prelude bootstrap
 
-stage0/bin/skargo: bootstrap/skargo_out64.ll stage0/lib/libskip_runtime64.a stage0/lib/libbacktrace.a
+stage0/bin/skc: bootstrap/skc_out64.ll bootstrap/libskip_runtime64.a bootstrap/libbacktrace.a
 	mkdir -p $(@D)
 	clang++ -no-pie $(CXXFLAGS) -o $@ $^ -lpthread -g3
 
-stage0/bin/skfmt: bootstrap/skfmt_out64.ll stage0/lib/libskip_runtime64.a stage0/lib/libbacktrace.a
+stage0/bin/skargo: bootstrap/skargo_out64.ll stage0/lib/libstd.sklib
```

Comment:
Just to check, the dependencies here are different because bootstrap/skc_out64.ll contains the bitcode not only of skc but also that of the stdlib and preamble, right? And this skc pattern is the reason that there are targets above for bootstrap/libskip_runtime64.a and bootstrap/libbacktrace.a separate from bootstrap/libstd.sklib?

Reply:
The dependencies are different here for bootstrap reasons: in order to generate a .sklib (which would contain the object files of `libskip_runtime64.a` and `libbacktrace.a`), we need `skc`, but in order to build `skc` from `skc_out64.ll`, we need `libskip_runtime64.a` and `libbacktrace.a`.

Reply:
Maybe what I really mean is, since there is bootstrap/libstd.sklib, couldn't stage0/bin/skc be built from that? That way skc.ll would not need to be expanded with the stdlib ll, and maybe bootstrap/libskip_runtime64.a bootstrap/libbacktrace.a would not be needed separate from bootstrap/libstd.sklib.

Reply:
And thinking of the scenario where we need to rebuild the runtime from C source but otherwise use the bootstrap, since sklib is in ar format, the bitcode inside could be extracted and used with freshly built native .a files. Not a super easy workflow, but possible and not something that we expect to do more than a few times ever.

Reply:
Ah, ok, I see. Nevermind my last comments then.